posthog-core/src/index.ts

```diff
@@ -0,0 +1,460 @@
+import {
+  PostHogFetchOptions,
+  PostHogFetchResponse,
+  PostHogQueueItem,
+  PostHogAutocaptureElement,
+  PostHogDecideResponse,
+  PosthogCoreOptions,
+  PostHogEventProperties,
+  PostHogPersistedProperty,
+} from './types'
+import { assert, currentISOTime, currentTimestamp, generateUUID, removeTrailingSlash, retriable } from './utils'
+export * as utils from './utils'
+import { LZString } from './lz-string'
+import { SimpleEventEmitter } from './eventemitter'
+
+export abstract class PostHogCore {
+  // options
+  private apiKey: string
+  private host: string
+  private flushAt: number
+  private flushInterval: number
+  private captureMode: 'form' | 'json'
+  private sendFeatureFlagEvent: boolean
+  private flagCallReported: { [key: string]: boolean } = {}
+
+  // internal
+  private _events: SimpleEventEmitter
+  private _queue: PostHogQueueItem[]
+  private _flushTimer?: any
+
+  private _decideResponsePromise?: Promise<PostHogDecideResponse>
+  private _decideResponse?: PostHogDecideResponse
+  private _decideTimer?: any
+  private _decidePollInterval = 10000
```

Comment:
Looks like we can't override this interval via `PosthogCoreOptions`?

Reply:
This is only temporarily hardcoded, I'll be adding it to the options for sure 👍